File: programming_hw_4s/tasks_with_eolymp_tags/t_13_2_(eolimp_6127).py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class LinkedQueue:
    def __init__(self):
        self.front = None
        self.size = 0

    def push(self, data):
        new_node = Node(data)

        if self.is_empty():
            self.front = new_node
        else:
            tmp = self.front
            while tmp.next is not None: tmp = tmp.next
            tmp.next = new_node
            del tmp

        self.size += 1
        return "ok"

    def pop(self):
        if self.is_empty(): return "error"
        front = self.front.data
        self.front = self.front.next
        self.size -= 1
        return front

    def get_front(self):
        if self.is_empty(): return "error"
        return self.front.data

    def get_size(self):
        return self.size

    def is_empty(self):
        return self.front is None

    def clear(self):
        if self.front:
            self.front = None
            self.size = 0
        return "ok"


new_linked_queue = LinkedQueue()


def procedure(cmd):
    global new_linked_queue

    if " " in cmd: return new_linked_queue.push(int(cmd.split()[1]))
    if cmd == "pop": return new_linked_queue.pop()
    if cmd == "front": return new_linked_queue.get_front()
    if cmd == "size": return new_linked_queue.get_size()
    if cmd == "clear": return new_linked_queue.clear()
    return ""


while True:
    command = input()
    if command == "exit":
        print("bye")
        break

    print(procedure(command))

File: programming_hw_4s/tasks_with_eolymp_tags/t_13_2_(eolimp_6127).cpp
#include <iostream>
#include <string>
using namespace std;

struct Node{
	Node * prev;
	int x;
	Node() { prev = nullptr; x = 0;}
	Node(Node * a, int b) { prev = a; x = b; }
};

struct queue{
private:
	Node * head = nullptr;
	Node * tail = nullptr;
	int count = 0;
public:
	int size() {
		return count;
	}

	void push(int p){
		if(count == 0){
			Node * NewNode = new Node(tail, p);

			head = NewNode;
			tail = NewNode;
		}
		else{
			Node * NewNode = new Node(nullptr, p);

			tail->prev = NewNode;
			tail = NewNode;
		}

		count++;
	}

	int front(){
		if(count) { return head->x; }
		return 0;
	}

	int pop(){
		if(count){
			int x = head->x;
			Node * NewNode = head->prev;

			delete head;

			head = NewNode;
			count--;

			return x;
		}
		return 0;
	}

	void clear(){
		while(count > 0){
			Node * NewNode = head->prev;

			delete head;

			head = NewNode;
			count--;
		}
		head = nullptr; tail = nullptr;
	}
};

int main() {
	queue new_queue;

	while(true){
		string s;
		cin >> s;

		if(s == "exit") {
			cout << "bye\n";
			break;
		}
		else if(s == "push"){
			int inp; cin >> inp; new_queue.push(inp); cout << "ok\n";
		}
		else if(s == "front"){
			if(new_queue.size()) cout << new_queue.front() << endl;
			else cout << "error\n";
		}
		else if(s == "size") {
			cout << new_queue.size() << endl;
		}
		else if(s == "pop"){
			if(new_queue.size()) cout << new_queue.pop() << endl;
			else cout << "error\n";
		}
		else if(s == "clear") {
			cout << "ok\n";
			new_queue.clear();
		}
	}
	return 0;
}